File: sem iv/Theory of Computaion/lab/code/c/1_substring_position.cpp
// Program: Extract substring given starting position (a) and ending position (b)
#include <iostream>
#include <cstring>
using namespace std;

void substring1(char w[100], int start, int end);

int main() {
    char input[100];
    int start, end;

    cout << "Enter the string" << endl;
    cin.getline(input, 100);

    do {
        cout << "Enter the starting position of the substring" << endl;
        cin >> start;
        cout << "Enter the ending position of the substring" << endl;
        cin >> end;
    } while (start < 1 || end > (int)strlen(input));

    substring1(input, start - 1, end - 1);
    cin.ignore();
    cin.get();
    return 0;
}

void substring1(char w[100], int start, int end) {
    cout << "The substring is" << endl;
    for (int i = start; i <= end; i++)
        cout << w[i];
    cout << endl;
}

File: sem iv/Theory of Computaion/lab/code/python/1_substring_position.py
# Program: Extract substring given starting position (a) and ending position (b)

def substring1(w, start, end):
    print("The substring is")
    print(w[start:end + 1])

def main():
    inp = input("Enter the string\n")
    while True:
        start = int(input("Enter the starting position of the substring\n"))
        end = int(input("Enter the ending position of the substring\n"))
        if not (start < 1 or end > len(inp)):
            break
    substring1(inp, start - 1, end - 1)

if __name__ == "__main__":
    main()
